src/api_callbacks.c

```diff
@@ -0,0 +1,139 @@
+#include "api_callbacks.h"
+#include "api_uart.h"
+#include "tools.h"
+#include <stdint.h>
+#include <string.h>
+#include <stdio.h>
+
+sensor_choice_t                         g_chosen_sensor = DEFAULT_COMMANDS;
+uint32_t                                g_command_parameters[MAX_COMMAND_PARAMETERS];
+uint8_t                                 g_command_param_counter = 0;
+static bmp280_data_s                    g_bmp_data = {0};
+static uint32_t                         g_sts32_val = 0;
+static measurement_config_t             g_sts32_config = REPEATABILITY_LOW;
+
+hub_retcode_t api_send_response(const uint32_t value)
+{
+    char msg[20] = {0};
+    sprintf(msg, "%lu", value);
+    return api_uart_tx((uint8_t *)msg, strlen(msg));
+}
+
+void set_irq_pin(void)
+{
+    HAL_GPIO_WritePin(IRQ_GPIO, IRQ_PIN, 0);
+}
+void clear_irq_pin(void)
+{
+    HAL_GPIO_WritePin(IRQ_GPIO, IRQ_PIN, 1);
+}
+
+void api_menu_set_bmp(void)
+{
+    g_chosen_sensor = BMP_COMMANDS;
+}
+void api_menu_set_sts(void)
+{
+    g_chosen_sensor = STS_COMMANDS;
+}
+void api_menu_set_mq(void)
+{
+    g_chosen_sensor = MQ_COMMANDS;
+}
+void api_menu_set_default(void)
+{
+    g_chosen_sensor = DEFAULT_COMMANDS;
+}
+
+void api_bmp_get_data(void)
+{
+    memset(&g_bmp_data, 0, sizeof(bmp280_data_s));
+    bmp280_get_data(&g_bmp_data);
+    api_send_response(g_bmp_data.temperature);
```

Comment:
bmp miał być do ciśnienia a koniec końców temperatury miał nie wysyłać

Reply:
Stanęło na tym, że uznajemy to po prostu za osobne czujniki i wystawiamy pełną funkcjonalność.